src/planner/agg_bookend.c

```diff
@@ -585,6 +585,11 @@ build_first_last_path(PlannerInfo *root, FirstLastAggInfo *fl_info, Oid eqop, Oi
 	subroot->parse = parse = copyObject(root->parse);
 	IncrementVarSublevelsUp((Node *) parse, 1, 1);
 
+#if PG16_GE
+	/* Reset placeholdersFrozen: https://github.com/postgres/postgres/commit/b3ff6c74 */
+	subroot->placeholdersFrozen = false;
+#endif
```

Comment:
you should add an explicit include for compat/compat.h to make sure PG16_GE is always defined

Reply:
Fixed.